container/root/etc/services-available/nginx/run

```diff
@@ -1,4 +1,18 @@
-#!/usr/bin/execlineb -P
-s6-setuidgid www-data
+#!/usr/bin/with-contenv bash
 
-nginx -g "daemon off;"
+# @see http://blog.tutum.co/2015/05/20/s6-made-easy-with-the-s6-overlay/ "Using signals besides TERM"
+
+# Nginx uses SIGQUIT to gracefully stop serving traffic.
+# Until S6 has a method to trap/replace those signals for docker-stop, trade signals here
+
+pid=$$
+
+sigterm() {
+kill -s SIGQUIT $pid
+}
+
+trap sigterm SIGTERM
```

Comment:
only trapping SIGTERM (from `docker stop`), as this is the only signal that would allow graceful shutdown.